api/src/wfl/module/covid.clj

```diff
@@ -157,34 +195,105 @@
 
 ;; Create and add new snapshots to the snapshot queue
 (defn ^:private update-tdr-source [source]
-  (letfn [(find-new-rows       [source now]      [])
-          (make-snapshots      [source row-ids])
-          (write-snapshots     [source snapshots])
+  (letfn [(find-new-rows [source now] [])
+          (make-snapshots [source row-ids])
+          (write-snapshots [source snapshots])
           (update-last-checked [now])]
     (let [now     (OffsetDateTime/now)
           row-ids (find-new-rows source now)]
       (when-not (empty? row-ids)
         (write-snapshots source (make-snapshots source row-ids)))
       (update-last-checked source now))))
 
-(defn ^:private load-tdr-source [{:keys [source-ptr] :as workload}]
-  {:type    tdr-source-type
-   :queue   nil
-   :updated nil})
+(defn ^:private load-tdr-source [tx {:keys [source_items] :as details}]
+  (if-let [source (->> (util/parse-int source_items)
+                       (conj ["SELECT * FROM TerraDataRepoSource WHERE id = ? LIMIT 1"])
+                       (jdbc/query tx)
+                       first)]
+    (assoc source :type tdr-source-type)
+    (throw (ex-info (str "No source matching id " source_items) details))))
 
 ;; maybe we want to split check-request and create to avoid writing
 ;; records for bad workload requests?
-(defn ^:private create-tdr-source [source-request]
-  (letfn [(check-request! [request] (throw (Exception. "Not implemented")))
-          (make-record    [request] nil)]
-    (check-request! source-request)
-    (let [foreign-key (make-record source-request)]
-      (load-tdr-source {:source-ptr foreign-key}))))
-
-(defoverload workloads/start-workload! pipeline start-covid-workload!)
-
-(defoverload create-source! tdr-source-type create-tdr-source)
-(defoverload peek-queue!    tdr-source-type peek-tdr-source-queue)
-(defoverload pop-queue!     tdr-source-type pop-tdr-source-queue)
+(defn ^:private create-tdr-source [tx id request]
+  (let [create    "CREATE TABLE %s OF TerraDataRepoSourceDetails (PRIMARY KEY (id))"
+        details   (format "%sDetails_%09d" tdr-source-type id)
+        _         (jdbc/execute! tx [(format create details)])
+        req->cols {:dataset :dataset
```

Comment:
I think this could be used both ways to rename the keys before we write to the database and when we read. It creates a nice separation between the in-memory and serialised forms. Consider extracting and using in both create and load implementations (and similarly with the others)